rdflib/plugins/sparql/sparql.py

```diff
@@ -264,11 +265,27 @@ def __init__(
         self.graph: Optional[Graph]
         self._dataset: Optional[ConjunctiveGraph]
         if isinstance(graph, ConjunctiveGraph):
-            self._dataset = graph
-            if rdflib.plugins.sparql.SPARQL_DEFAULT_GRAPH_UNION:
-                self.graph = self.dataset
+            if datasetClause:
+                self._dataset = ConjunctiveGraph()
+                self.graph = Graph()
+                for d in datasetClause:
+                    if d.default:
+                        self.graph += graph.get_context(d.default)
+                        if not graph.get_context(d.default):
+                            self.load(d.default, default=True)
```

Comment:
Why call `graph.get_context(d.default)` twice?
```suggestion
                       from_graph = graph.get_context(d.default)
                       self.graph += from_graph
                        if not from_graph:
                            self.load(d.default, default=True)
```

Reply:
Thanks for your suggestion @ashleysommer, I also did the same for the named graphs